Cargo.toml

```diff
@@ -211,7 +211,7 @@ diplomat-tool = { git = "https://github.com/rust-diplomat/diplomat.git", rev = "
 
 # External deps from crates.io
 bincode = { version = "1.3.1", default-features = false }
-clap = { version = "4.2.0", default-features = false }
+clap = { version = "4.2.0" } # clap has a lot of default features and we want them
```

Comment:
can we not do this globally? enable the default feature in the crates that want them.

Reply:
I don't think we'd ever want clap without its default features, which largely enable diagnostics and help text and things, but ok

Reply:
Okay, I made this change, but I'm not convinced it's an improvement. It would be nice if we never had to rebuild the clap dependency when different icu4x projects use it with different features. Why allow projects to use clap with a different set?

Reply:
Yeah I'm fine with doing this globally.

Reply:
There's no diff in `Cargo.lock`, does this not pull in any additional dependencies?

Reply:
There are other deps that pull in clap, such as `criterion` and `diplomat-tool`

Reply:
- @sffc - We don't want to take on no-default-features liability.
- @Manishearth - Our no-default-features deps for code library crates are either tiny or maintained by BurntSushi who follows an even stricter version of the no-default-features stability policy.
- @robertbastian - If we are explicit about why deps have different features in Cargo.toml (in groups like allowlist.rs) it's fine with me.